### PR TITLE
BaSP-MR-37: test super admins get and post and HOTFIX regex password

### PR DESCRIPTION
**Pull request by Micaela-Rossi**
Create a seed for super admins, and create tests for get, get by id and post.
Also made some modification to the validations to repair some errors with the password regex.

### CODE REVIEW

File: src/tests/super-admins.test.js [hunk 1]
```diff
@@ -1 +1,162 @@
...
+    const response = await request(app).post('/api/superAdmins').send();
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBeUndefined();
+  });
+  test('When create an super admin status must be 201, and must have firstName, email and password defined', async () => {
+    const response = await request(app).post('/api/super-admins').send(mockSuperAdmin);
+    expect(response.status).toBe(201);
+    expect(response.body).toBeDefined();
+    expect(response.body.data).toHaveProperty('firstName', 'email', 'password');
+    expect(response.body.data.email).toMatch(/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.+-]+\.com$/);
```
[the remark this answers is not in the document]
Comment (francoax) [reply src/tests/super-admins.test.js:123]: For me, if you do a test before where you are testing that the data is sent correctly, here is not necessary to do it

**Review by francoax — CHANGES_REQUESTED** [review]

File: src/tests/super-admins.test.js [hunk 2]
```diff
@@ -1 +1,162 @@
...
+    expect(response.body.message).toMatch('Super Admin Created!');
+    expect(response.body.error).toBeFalsy();
+  });
+  test('When create an super admin status must be 400 if name is not defined', async () => {
```
Comment (Eliezer-Alberto) [inline src/tests/super-admins.test.js:127]: There is a 'n' left over in "an".. the same happens in some following lines.. 
'a' instead of 'an'.

**Review by Eliezer-Alberto — CHANGES_REQUESTED** [review]

File: src/tests/super-admins.test.js [hunk 3]
```diff
@@ -1 +1,162 @@
...
+    expect(response.body).toBeDefined();
+    expect(response.body.message).toMatch('Super Admins list: ');
+  });
+  test('The status must be 404, if the url is wrong', async () => {
+    const response = await request(app).get('/api/superAdmins').send();
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBeUndefined();
+  });
+  describe('GET /api/super-admins', () => {
+    test('The database must return 404 if it\'s empty of super admins. An said Cannot find any Super Admin, please create one.', async () => {
```
Comment (Eliezer-Alberto) [inline src/tests/super-admins.test.js:72]: There is a 'd' missing in 'and'..

**Review by Eliezer-Alberto — CHANGES_REQUESTED** [review]

**Review by Eliezer-Alberto — APPROVED** [review]

File: src/tests/super-admins.test.js [hunk 4]
```diff
@@ -1 +1,150 @@
...
+    expect(response.status).toBe(200);
+    expect(response.body).toBeDefined();
+    expect(response.body.data).toHaveProperty('_id', 'firstName', 'email', 'password');
+    expect(response.body.message).toMatch(`Super Admin Found! ${superAdminSeed[0].firstName}`);
+    expect(response.body.error).toBeFalsy();
+  });
+  test('The status must be 404, if the url is wrong', async () => {
+    const response = await request(app).get('/api/superAdmins').send();
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBeUndefined();
```
Comment (Dannylez) [inline src/tests/super-admins.test.js:82]: be careful, in these cases the response doesn't have a ".body.error" so it doesn't make sense to test it for undefined

**Review by nailaghigo — APPROVED** [review]
🚀

File: src/tests/super-admins.test.js (hunk 4 above)
Comment (Micaela-Rossi) [reply src/tests/super-admins.test.js:82]: Nai corrected these like that cause i told her the body error doesn't match to falsy or thruty. The test throws undefined

**Pull request merged**